### ISSUE TITLE
Invalid input type: Batch norm layer expected input of type CNN...

### ISSUE THREAD

**Issue by Corgile** [deeplearning4j/deeplearning4j#9967]
### Java code

```java
String simpleMlp = new ClassPathResource("m2_128.h5").getFile().getPath();
System.out.println( simpleMlp);
InputStream inputStream = new FileInputStream(simpleMlp);
MultiLayerNetwork model = KerasModelImport.importKerasSequentialModelAndWeights(inputStream, false);
```
### output (using diff for highlight)
```diff
- SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
- SLF4J: Defaulting to no-operation (NOP) logger implementation
- SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
/home/gzhuadmin/workspace/dl4j/target/classes/m2_128.h5
- Exception in thread "main" java.lang.IllegalStateException: Invalid input type: Batch norm layer expected input of type CNN, CNN Flat or FF, got InputTypeRecurrent(64,timeSeriesLength=12,format=NWC) for layer index -1, layer name = batch_normalization
	at org.deeplearning4j.nn.conf.layers.BatchNormalization.getOutputType(BatchNormalization.java:131)
	at org.deeplearning4j.nn.modelimport.keras.layers.normalization.KerasBatchNormalization.getOutputType(KerasBatchNormalization.java:165)
	at org.deeplearning4j.nn.modelimport.keras.KerasModel.inferOutputTypes(KerasModel.java:304)
	at org.deeplearning4j.nn.modelimport.keras.KerasSequentialModel.<init>(KerasSequentialModel.java:146)
	at org.deeplearning4j.nn.modelimport.keras.KerasSequentialModel.<init>(KerasSequentialModel.java:57)
	at org.deeplearning4j.nn.modelimport.keras.utils.KerasModelBuilder.buildSequential(KerasModelBuilder.java:322)
	at org.deeplearning4j.nn.modelimport.keras.KerasModelImport.importKerasSequentialModelAndWeights(KerasModelImport.java:208)
	at org.deeplearning4j.nn.modelimport.keras.KerasModelImport.importKerasSequentialModelAndWeights(KerasModelImport.java:94)
	at iot.xhl.Main.main(Main.java:58)

```

## Version Information

Please indicate relevant versions, including, if relevant:

### Deeplearning4j version

<img width="555" alt="image" src="https://user-images.githubusercontent.com/48396880/236252664-efc400e2-398d-48d1-8d2a-7c363dc8e881.png">

### Platform information (OS, etc)

IDEA,  Ubuntu2004, open-JDK-11

Keras 2.12.0  TensorFlow 2.12.0

### CUDA version, if used

not using CUDA

## Additional Information

Where applicable, please also provide:
### full pom.xml file

<script src="https://gist.github.com/Corgile/ee6ada2ec050664c41698a1a507b6fa0.js"></script>

link here

https://gist.github.com/Corgile/ee6ada2ec050664c41698a1a507b6fa0

### model stuff

```python
   #Bidirectional RNN
from keras.models import Sequential
...

batch_size = 32
model = Sequential()
model.add(Convolution1D(64, kernel_size=64, padding="same",activation="relu",input_shape=(32, 1)))
model.add(MaxPooling1D(pool_size=(10)))
model.add(BatchNormalization())
model.add(Bidirectional(LSTM(64, return_sequences=False))) 
model.add(Reshape((128, 1), input_shape = (128, )))

model.add(MaxPooling1D(pool_size=(5)))
model.add(BatchNormalization())
model.add(Bidirectional(LSTM(128, return_sequences=False))) 

model.add(Dropout(0.6))
model.add(Dense(8))
model.add(Activation('softmax'))
model.compile(loss='categorical_crossentropy',optimizer='adam',metrics=['accuracy'])

...

model.summary()
```

```log
Model: "sequential"
_________________________________________________________________
 Layer (type)                Output Shape              Param #   
=================================================================
 conv1d (Conv1D)             (None, 32, 64)            4160      
                                                                 
 max_pooling1d (MaxPooling1D  (None, 3, 64)            0         
 )                                                               
                                                                 
 batch_normalization (BatchN  (None, 3, 64)            256       
 ormalization)                                                   
                                                                 
 bidirectional (Bidirectiona  (None, 128)              66048     
 l)                                                              
                                                                 
 reshape (Reshape)           (None, 128, 1)            0         
                                                                 
 max_pooling1d_1 (MaxPooling  (None, 25, 1)            0         
 1D)                                                             
                                                                 
 batch_normalization_1 (Batc  (None, 25, 1)            4         
 hNormalization)                                                 
                                                                 
 bidirectional_1 (Bidirectio  (None, 256)              133120    
 nal)                                                            
                                                                 
 dropout (Dropout)           (None, 256)               0         
                                                                 
 dense (Dense)               (None, 8)                 2056      
                                                                 
 activation (Activation)     (None, 8)                 0         
                                                                 
=================================================================
Total params: 205,644
Trainable params: 205,514
Non-trainable params: 130
_________________________________________________________________

```

### How I saved my model

<img width="430" alt="image" src="https://user-images.githubusercontent.com/48396880/236255356-0fd03cb3-3b6a-4cea-ad3c-bcaf68ba9c3f.png">

**Comment by Corgile** [comment]
if you want to check my project, please send me an email. :)

**Comment by agibsonccc** [comment]
@Corgile I'm not supporting a 2 year old version of the project. Update and I'll re open the issue if it's still relevant.

**Comment by Corgile** [comment]
> @Corgile I'm not supporting a 2 year old version of the project. Update and I'll re open the issue if it's still relevant.

Hello @agibsonccc ,  I update all the dependencies to the latest release aka `1.0.0-M2.1` Here are the new error messages:

(after `mvn clean install`)

<img width="1188" alt="image" src="https://user-images.githubusercontent.com/48396880/236373826-eaa2fdea-6223-4a52-9471-0763b6b34da4.png">


full log

```log
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
/home/gzhuadmin/workspace/dl4j/target/classes/m2_128.h5
HDF5-DIAG: Error detected in HDF5 (1.12.1) thread 0:
  #000: H5G.c line 510 in H5Gopen2(): unable to open group
    major: Symbol table
    minor: Can't open object
  #001: H5VLcallback.c line 4187 in H5VL_group_open(): group open failed
    major: Virtual Object Layer
    minor: Can't open object
  #002: H5VLcallback.c line 4154 in H5VL__group_open(): group open failed
    major: Virtual Object Layer
    minor: Can't open object
  #003: H5VLnative_group.c line 123 in H5VL__native_group_open(): unable to open group
    major: Symbol table
    minor: Can't open object
  #004: H5Gint.c line 270 in H5G__open_name(): group not found
    major: Symbol table
    minor: Object not found
  #005: H5Gloc.c line 442 in H5G_loc_find(): can't find object
    major: Symbol table
    minor: Object not found
  #006: H5Gtraverse.c line 837 in H5G_traverse(): internal path traversal failed
    major: Symbol table
    minor: Object not found
  #007: H5Gtraverse.c line 613 in H5G__traverse_real(): traversal operator failed
    major: Symbol table
    minor: Callback failed
  #008: H5Gloc.c line 399 in H5G__loc_find_cb(): object 'lstm_cell_1' doesn't exist
    major: Symbol table
    minor: Object not found
Exception in thread "main" java.lang.RuntimeException: H5Gopen2 failed
	at org.bytedeco.hdf5.H5Location.openGroup(Native Method)
	at org.deeplearning4j.nn.modelimport.keras.Hdf5Archive.openGroups(Hdf5Archive.java:80)
	at org.deeplearning4j.nn.modelimport.keras.Hdf5Archive.readDataSet(Hdf5Archive.java:108)
	at org.deeplearning4j.nn.modelimport.keras.utils.KerasModelUtils.importWeights(KerasModelUtils.java:329)
	at org.deeplearning4j.nn.modelimport.keras.KerasSequentialModel.<init>(KerasSequentialModel.java:154)
	at org.deeplearning4j.nn.modelimport.keras.KerasSequentialModel.<init>(KerasSequentialModel.java:55)
	at org.deeplearning4j.nn.modelimport.keras.utils.KerasModelBuilder.buildSequential(KerasModelBuilder.java:326)
	at org.deeplearning4j.nn.modelimport.keras.KerasModelImport.importKerasSequentialModelAndWeights(KerasModelImport.java:203)
	at org.deeplearning4j.nn.modelimport.keras.KerasModelImport.importKerasSequentialModelAndWeights(KerasModelImport.java:89)
	at iot.xhl.Main.main(Main.java:58)

```

I looked up the related issues by searching `HDF5-DIAG` in issue tab just now,  but none of their solutions works for me, So I'm reopening this issue, :semi

**Comment by agibsonccc** [comment]
That looks like it could be an invalid file? That is actually an hdf5 error (read: not maintained by us) could you research that a bit to be sure? I would also accept seeing if https://netron.app can open it

**Comment by Corgile** [comment]
> That looks like it could be an invalid file? That is actually an hdf5 error (read: not maintained by us) could you research that a bit to be sure? I would also accept seeing if https://netron.app can open it

Yes, it can

<img width="294" alt="image" src="https://user-images.githubusercontent.com/48396880/236375764-f6ab5577-a7df-4e49-8aff-e238b672c068.png">

**Comment by agibsonccc** [comment]
Ok DM me the model on the community forums https://community.konduit.ai/ thanks!
